matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -214,21 +223,25 @@ async fn message_loop<M: Messenger>(
                             .get_mut(&peer)
                             .expect("couldn't find data channel for peer")
                             .get_mut(channel_index).unwrap_or_else(|| panic!("couldn't find data channel with index {channel_index}"));
-                        data_channel.send(packet).unwrap();
-
+                        if let Err(e) = data_channel.send(packet) {
+                            // Peer we're sending to closed their end of the connection.
+                            // We anticipate the PeerLeft event soon, but we sent a message before it came.
+                            // Do nothing. Only log it.
+                            warn!("failed to send to peer {peer} (socket closed): {e:?}");
+                        };
                     }
                     Some((_, None)) | None => {
                         // Receiver end of outgoing message channel closed,
                         // which most likely means the socket was dropped.
                         // There could probably be cleaner ways to handle this,
                         // but for now, just exit cleanly.
-                        debug!("Outgoing message queue closed");
-                        break;
+                        warn!("Outgoing message queue closed, message not sent");
+                        break Ok(());
```

Comment:
I feel like this should probably result in some kind of error being raised, perhaps a `SignalingError::SocketDropped`. In my head this isn't part of the expected messaging loop life-cycle and is thus erroneous behaviour

Reply:
I changed it to break with `SignalingError::StreamExhausted` which is the current going alias for `SocketDropped`. This is to keep verbage consistent with WsStream, since they call it `SteamExhausted`.